File: samples/go/03sqrt/sqrt.go
package main

import (
    "fmt"
    "os"
    "strconv"
)


func findSqrt(number float64) (float64) {
    var x float64 = float64(number);
    var y float64= 1.0;
    const e float64 = 0.0000001;
    for x - y > e {
        x = (x + y) / 2.0;
        y = number / x;
    }
    return x;
}


func main() {
    if (len(os.Args) < 2) {
        fmt.Fprintln(os.Stderr, "Error: Usage ./sqrt <number> [<number> ...]");
        os.Exit(1);
    }

    for _, arg := range os.Args[1:] {
        var number int = 0;
        number, err := strconv.Atoi(arg);
        if err != nil {
            fmt.Printf("Error: Expected to receive an integer, received %s\n", arg);
            os.Exit(2);
        }
        fmt.Printf("%d => %f\n", number, findSqrt(float64(number)))
    }
}

File: samples/rust/03sqrt/sqrt.rs
use std::env;

static ACCURACY: f32 = 0.000001;

fn find_sqrt(num: f32) -> f32 {
    let mut x: f32 = num;
    let mut y: f32 = 1.0;
    while x - y > ACCURACY {
        x = (x+y)/2.0;
        y = num/x;
    }
    return x;
}

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() == 1 {
        eprintln!("Error: Usage {} <number> [<number> ...]", args[0]);
        std::process::exit(1);
    }

    for arg in &args[1 .. args.len()] {
        let result = arg.parse::<f32>();
        match result {
            Ok(v) if v > 0.0 => println!("{} => {}", v, find_sqrt(v)),
            Ok(v) => {
                eprintln!("Error: Expected to receive a positive number. Received {}", v);
                std::process::exit(3);
            },
            Err(_e) => {
                eprintln!("Error: Expected to receive a number. Received {}", arg);
                std::process::exit(2);
            }
        }
    }
}
